src/main/java/seedu/address/logic/commands/FindDeliveryCommand.java

```diff
@@ -59,7 +59,7 @@ public CommandResult execute(Model model) {
 
         // If supplier index is provided, retrieve the supplier and filter by that supplier
         if (supplierIndex.isPresent()) {
-            List<Supplier> lastShownList = model.getFilteredSupplierList();
+            List<Supplier> lastShownList = model.getModifiedSupplierList();
```

Comment:
Nice catch. We should do one more round of check to ensure getModifiedSupplierList is used properly